### PR TITLE
feat: add pagesDir glob

### PR DESCRIPTION
**Pull request by klarkc**
With this, `pagesDir` can receive a glob string, so one can set multiple `pages` directories using globs instead of fixed arrays

### CODE REVIEW

**Comment by hannoeru** [comment]
Can you use `resolveOptions()` to resolve glob pages and merge it with existing options?

File: test/__snapshots__/files.spec.ts.snap [hunk 1]
```diff
@@ -14,3 +14,16 @@ Array [
   "index.vue",
 ]
 `;
+
+exports[`Get page dirs Page dirs with glob pagesDir: pages dirs with glob pagesDir 1`] = `
+Array [
+  Object {
+    "baseRoute": "/",
+    "dir": "/home/klarkc/Sources/vite-plugin-pages/test/assets/deep-pages/bar/pages",
```
Comment (hannoeru) [inline test/__snapshots__/files.spec.ts.snap:22]: This will failed in other env.

**Comment by klarkc** [comment]
> Can you use `resolveOptions()` to resolve glob pages and merge it with existing options?

Can't because plugin initialization is sync, but I moved it to resolveConfig hook which is async

**Comment by hannoeru** [comment]
Thanks, that is great!

**Pull request merged**